src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -69,6 +69,19 @@ public AcquireTokenForClientParameterBuilder WithSendX5C(bool withSendX5C)
             return this;
         }
 
+        /// <summary>
+        /// Specifies to use the Managed Identity to fetch the token from the Managed Identity Endpoint.
+        /// </summary>
+        /// <returns>The builder to chain the .With methods</returns>
+        public AcquireTokenForClientParameterBuilder WithManagedIdentity()
+        {
+            ValidateUseOfExperimentalFeature("ManagedIdentity");
+
+            ServiceBundle.Config.UseManagedIdentity = true;
```

Comment:
Can you review all other existing APIs and see if they make sense when used in conjunction with MSI feature? 

I'm thinking that stuff like `AppTokenProvider`, `Credential`, `Regional`, anything related to instance discovery doesn't make sense here. For some we can ignore them, for others we want to throw exception.

Reply:
Reviewed other APIs and added in the api review doc. Based on the discussion there I will make changes here.